packages/java-edition/src/binder/index.ts

```diff
@@ -178,18 +178,14 @@ export function dissectUri(uri: string, ctx: UriBinderContext) {
 			// Try to match the category for custom resources.
 			const subFolders = match[3].split('/').slice(0, -1)
 
-			if (subFolders.length === 0) {
-				continue
-			} else {
-				for (let i = 0; i < subFolders.length - 1; i++) {
-					const customCategory = `${match[2]}/${
-						subFolders.slice(0, i + 1).join('/')
-					}`
-
-					def = Categories.get(customCategory)
-
-					// we don't break here because we want to find the most specific category
-				}
+			for (let i = 0; i < subFolders.length - 1; i++) { // if subFolders is empty, this loop will not run
+				const customCategory = `${match[2]}/${
+					subFolders.slice(0, i + 1).join('/')
+				}`
+
+				def = Categories.get(customCategory)
```

Comment:
wait. isn't this loop ultimately just the same as doing this?

```ts
const customCategory = `${match[2]}/${subFolders.join('/')}`
def = Categories.get(customCategory)
```

we're never doing anything with the `def` assignment mid-loop so it will functionally only ever do stuff with the final iteration of the loop